kernel/arch/dreamcast/include/dc/syscalls.h

```diff
@@ -0,0 +1,395 @@
+/* KallistiOS ##version##
+
+   dc/syscalls.h
+   Copyright (C) 2024 Andy Barajas
+*/
+
+/** \file      dc/syscalls.h
+    \brief     Functions to access the system calls of the Dreamcast ROM.
+    \ingroup   system_calls
+
+    \author Marcus Comstedt
+    \author Andy Barajas
+*/
+
+/** \defgroup  system_calls System Calls
+    \brief     API for the Dreamcast's system calls
+    \ingroup   system
+
+    This module encapsulates all the system calls available in the Dreamcast 
+    Operating System, allowing direct interaction with system hardware 
+    components such as the GDROM drive, flash ROM, and bios fonts. These 
+    functions are essential for performing low-level operations that are not
+    handled by standard user-space APIs.
+*/
+
+#ifndef __DC_SYSCALLS_H
+#define __DC_SYSCALLS_H
+
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
+#include <stdint.h>
+#include <sys/types.h>
+
+/** \brief   Reads an icon from the flashrom.
+    \ingroup system_calls
+
+    This function reads an icon from the flashrom into a destination 
+    buffer.
+
+    \note
+    The format of these icons is not known.
+
+    \param  icon            The icon number (0-9, 5-9 seems to really 
+                            be icons).
+    \param  dest            The destination buffer (704 bytes in size).
+
+    \return                 Number of bytes read on success, or -1 on
+                            failure.
+*/
+int syscall_sysinfo_icon(uint32_t icon, uint8_t *dest);
```

Comment:
I dunno, "number of bytes or size" is literally what the `size_t` type is for, and it returning a 0 for that data type is an obvious error... I feel like the return type shouldn't be an `int` here.